sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -59,11 +60,19 @@ object CostBasedJoinReorder extends Rule[LogicalPlan] with PredicateHelper {
   private def reorder(plan: LogicalPlan, output: Seq[Attribute]): LogicalPlan = {
     val (items, conditions) = extractInnerJoins(plan)
     val result =
-      // Do reordering if the number of items is appropriate and join conditions exist.
+      // Do reordering if the join conditions exist.
       // We also need to check if costs of all items can be evaluated.
-      if (items.size > 2 && items.size <= conf.joinReorderDPThreshold && conditions.nonEmpty &&
+      if (items.size > 2 && conditions.nonEmpty &&
           items.forall(_.stats.rowCount.isDefined)) {
-        JoinReorderDP.search(conf, items, conditions, output)
+        // Reorder with DP when the the number of items is appropriate,
+        // otherwise try GA if it is enabled.
+        if (items.size <= conf.joinReorderDPThreshold) {
+          JoinReorderDP.search(conf, items, conditions, output)
+        } else if (conf.joinReorderGAEnabled) {
+          JoinReorderGA.search(conf, items, conditions, output).getOrElse(plan)
+        } else {
+          plan
+        }
       } else {
         plan
       }
```

Comment:
nit. Eliminate two `else` here and return `plan` at the end of this function.

Reply:
Fixed.